packages/xr/src/lib/components/Controller.svelte

```diff
@@ -0,0 +1,117 @@
+  import { fire } from '../internal/events'
+  import type { XRController, XRControllerEvent } from '../types'
+  import { isHandTracking, activeTeleportController, pendingTeleportDestination } from '../internal/stores'
+  import { left, right, gaze } from '../hooks/useController'
+  import ControllerModel from '../components/ControllerModel.svelte'
+  import ShortRay from '../components/ShortRay.svelte'
+
+  const xrEvents = [
+    'select',
+    'selectstart',
+    'selectend',
+    'squeeze',
+    'squeezeend',
+    'squeezestart'
+  ] as const
+
+  </script>
+
+  <script lang='ts'>
+
+  export let index: number
+
+  type $$Events = {
+    connected: XRControllerEvent<'connected'>
+    disconnected: XRControllerEvent<'disconnected'>
+    select: XRControllerEvent<'select'>
+    selectstart: XRControllerEvent<'selectstart'>
+    selectend: XRControllerEvent<'selectend'>
+    squeeze: XRControllerEvent<'squeeze'>
+    squeezeend: XRControllerEvent<'squeezeend'>
+    squeezestart: XRControllerEvent<'squeezestart'>
+  }
+
+  const dispatch = createRawEventDispatcher<$$Events>()
+  const { xr } =  useThrelte().renderer
+  const controller = xr.getController(index)
+  const grip = xr.getControllerGrip(index)
+
+  let connected = false
+
+  const handleConnected = (event: XRControllerEvent<'connected'>) => {
+    const data = event.data!
+    const xrController: XRController = { controller, grip, inputSource: data }
+
+    connected = true
+
+    fire('connected', event)
+    dispatch('connected', event)
+
+    switch (data.handedness) {
+      case 'left': return left.set(xrController)
+      case 'right': return right.set(xrController)
+      case 'none': return gaze.set(xrController)
+    }
+  }
+
+  const handleDisconnected = (event: XRControllerEvent<'disconnected'>) => {
+    const data = event.data!
+
+    connected = false
+
+    fire('disconnected', event)
+    dispatch('disconnected', event)
+
+    switch (data.handedness) {
+      case 'left': return left.set(undefined)
+      case 'right': return right.set(undefined)
+      case 'none': return gaze.set(undefined)
+    }
+  }
+
+  const handleXrEvent = (event: XRControllerEvent<typeof xrEvents[number]>) => {
+    fire(event.type, event)
+    dispatch(event.type, event)
+  }
+
+  onMount(() => {
+    controller.addEventListener('connected', handleConnected)
+    controller.addEventListener('disconnected', handleDisconnected)
+    xrEvents.forEach((event) => controller.addEventListener(event, handleXrEvent))
+  })
+
+  onDestroy(() => {
+    controller.removeEventListener('connected', handleConnected)
+    controller.removeEventListener('disconnected', handleDisconnected)
+    xrEvents.forEach((event) => controller.removeEventListener(event, handleXrEvent))
+  })
+</script>
+
+<T
+  is={grip}
+  name='XR Controller Grip {index}'
+  visible={connected && !$isHandTracking}
+>
+  <slot>
+    <ControllerModel {index} />
+  </slot>
```

Comment:
This also seems to break the <TeleportControls> example. The default controllers are not visible anymore.

Reply:
I've been thinking a bit about `<Controllers>` ever since you raised this issue. I'm thinking about switching things up a bit in this PR, curious for your thoughts:

Maybe it makes sense to get rid of `<Controllers>` and just export `<Controller>`. 

Pros:

* Only 3 possible slots per `<Controller>` vs 6 possible slots for `<Controllers>`:
```svelte
<Controller handedness='left'>
  <T is={$gltf1} /> <!-- default, model -->
  <T slot='grip-space' is={$gltf2} />
  <T slot='controller-space' is={$gltf3} />
</Controller>
```

* No need to filter events, it's weird / uncommon currently that all events will fire for either controller:
```svelte
<Controller
  handedness='left'
  on:select={() => /* no need to filter by handedness like with <Controllers> */}
/>

* Fewer components to maintain

Cons:

* This architecture introduces a little complexity internally, since the only way to grab a reference to a controller object is by index and then wait for a connection to determine handedness